### PR TITLE
restructure releases page

### PR DESCRIPTION
**Pull request by imatwawana**
- Moves hero image further up the page
- Rearranges headings below the image
- Adds generic wording for code mapping files
- Adds Learn More heading
- Points to Set Up page instead of SDK pages for setup content

**Merge after [PR 5108](https://github.com/getsentry/sentry-docs/pull/5108)**

### CODE REVIEW

**Review by scttcper — APPROVED** [review]
"or other mapping files" still feels a bit weird, but we don't have a better word for it

**Comment by imatwawana** [comment]
> "or other mapping files" still feels a bit weird, but we don't have a better word for it

I probably won't merge this until Wednesday, so if you think of anything before then, feel free to circle back with the suggestion.

File: src/docs/product/releases/index.mdx
Comment (imatwawana) [inline src/docs/product/releases/index.mdx:13]:
```suggestion
- Each release version (a short version of the release name without the hash)
```

File: src/docs/product/releases/index.mdx
Comment (imatwawana) [inline src/docs/product/releases/index.mdx:21]:
```suggestion

Notifying Sentry of a release enables auto discovery of which commits to associate with a release and identifies what we consider "the most recent release" when searching in [sentry.io](https://sentry.io).
```

File: src/docs/product/releases/index.mdx
Comment (imatwawana) [inline src/docs/product/releases/index.mdx:37]:
```suggestion
You don't need a repository integration for any of these features, though we recommend [installing one as part of an automated release option](/product/releases/setup/release-automation/) for efficiency.


## Apply Source Maps or Mapping Files {#apply-source-maps}
```

**Review by scttcper — APPROVED** [review]

File: src/docs/product/releases/index.mdx
Comment (imatwawana) [inline src/docs/product/releases/index.mdx:23]:
```suggestion

Each release links to one or more projects. If a release has multiple projects, Sentry will duplicate the release data in relation to each one. From this page, you can also click any release to go to [Release Details](/product/releases/release-details/) for more information.
```

File: src/docs/product/releases/index.mdx
Comment (imatwawana) [inline src/docs/product/releases/index.mdx:37]: You don't need a repository integration for any of these features, though we recommend [installing one as part of an automated release option](/product/releases/setup/release-automation/) for efficiency.

File: src/docs/product/releases/index.mdx
Comment (imatwawana) [inline src/docs/product/releases/index.mdx:43]:
```suggestion
Releases are required to apply code mapping files, such as [source maps](/platforms/javascript/sourcemaps/) for our JavaScript SDK, so that you can view source code context obtained from stack traces in their original untransformed form. This is useful for debugging minified code (for example, UglifyJS), or transpiled code from a higher-level language (such as TypeScript and ES6).
```

**Pull request merged**
